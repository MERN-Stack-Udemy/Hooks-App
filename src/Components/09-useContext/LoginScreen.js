import React, { useContext } from 'react'
import { useForm } from '../../Hooks/useForm';
import { UserContext } from './UserContext';

export const LoginScreen = () => {

  const [ {userName}, handleChange ] = useForm({});

  const {setUser} = useContext(UserContext);

  const handleSubmit = (e) =>{
    e.preventDefault();
    setUser({
      id: new Date().getTime(),
      name: userName
    })
  }

  return (
    <div>
      <h1>Login Screen</h1>
      <hr/>
      <div className="container-md">
        <form onSubmit={ handleSubmit }>
          <input
            autoComplete="off"
            className="form-control"
            name="userName"
            onChange={handleChange}
            placeholder="User name"
            type="text"
            value={userName}
          />

          <button 
            className="btn btn-outline-primary mt-2 btn-block" 
            type="submit"
          >
            Login
          </button>
        </form>
      </div>
    </div>
  )
}
