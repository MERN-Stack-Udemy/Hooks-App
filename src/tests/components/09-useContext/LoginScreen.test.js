import { mount } from "enzyme"
import { LoginScreen } from "../../../Components/09-useContext/LoginScreen"
import { UserContext } from "../../../Components/09-useContext/UserContext"

describe('test on <LoginScreen/> component', () => {
    
    const setUser = jest.fn();
    const wrapper = mount(
        <UserContext.Provider value={{setUser}}>
            <LoginScreen/>
        </UserContext.Provider>
    )

    test('should be math to snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    })

    test('should call \'setUser\' with argument', () => {
        wrapper.find('button').simulate('submit');
        expect( setUser ).toHaveBeenCalledWith({id: 123, name: 'smog'});
    })
    
    
})
