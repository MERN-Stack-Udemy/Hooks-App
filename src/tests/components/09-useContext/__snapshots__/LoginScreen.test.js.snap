// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <LoginScreen/> component should be math to snapshot 1`] = `
<div>
  <h1>
    Login Screen
  </h1>
  <hr />
  <div
    className="container-md"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        autoComplete="off"
        className="form-control"
        name="userName"
        onChange={[Function]}
        placeholder="User name"
        type="text"
      />
      <button
        className="btn btn-outline-primary mt-2 btn-block"
        type="submit"
      >
        Login
      </button>
    </form>
  </div>
</div>
`;
